[feature] Review, Tag 엔터티 추가

ghrltjdtprbs opened the pull request:
### [feature] Review, Tag 엔터티 추가
- 하나의 리뷰에 여러 개의 태그가 있고, 각 태그도 여러 개의 리뷰에 사용될 수 있기 때문에 ManyToMany 로 설정
   -> 중간테이블 ReviewTag 생성
- 태그 중복을 허용하지 않기 때문에 Set으로 받아옴
- rating 유효성 검사
- Review 도메인 Flyway 스크립트 추가

ghrltjdtprbs committed:
[feat] Review 엔터티 추가

변경 사항 :
 - Review 엔터티 추가

ghrltjdtprbs committed:
[feat] Tag 엔터티 추가

변경 사항 :
 - Tag 엔터티 추가

JeongUijeong commented on src/main/java/com/fasttime/domain/review/entity/Review.java:
```diff
@@ -0,0 +1,48 @@
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+@Entity
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+public class Review extends BaseTimeEntity {
```
@Getter 가 필요할 것 같아요 ㅎㅎ

ghrltjdtprbs replied:
피드백 감사합니다. 수정하겠습니다! ㅎㅎ😊

JeongUijeong commented on src/main/java/com/fasttime/domain/review/entity/Tag.java:
```diff
@@ -0,0 +1,25 @@
+import java.util.Set;
+import lombok.NoArgsConstructor;
+
+@Entity
+@NoArgsConstructor
+public class Tag {
```
@Getter 가 필요할 것 같고, @NoArgsConstructor에 접근레벨 설정 해주시면 좋을 것 같습니다. 😄

ghrltjdtprbs replied:
수정하여 커밋했습니다! 확인해주세요 ㅎㅎ

JeongUijeong reviewed (changes requested):
수고하셨습니다! 😊 몇 가지 코멘트 남겼으니 참고하셔서 수정 부탁드립니다!
더불어 줄이 2줄 띄어져 있는 부분도 한 줄 공백으로 수정해주시면 감사하겠습니다! 🤍

Nine-JH reviewed (changes requested):
`@ManyToMany` 보다는 중간테이블을 하나 만드는게 더 좋아보이네요!

ghrltjdtprbs committed:
[refactor] Tag엔터티 접근레벨 설정

변경 사항 : 
 - @NoArgsConstructor에 접근 레벨 설정

ghrltjdtprbs committed:
[refactor] Review엔터티 공백제거

변경 사항 : 
 - 가독성을 위해 불필요한 공백을 제거하였습니다.

ghrltjdtprbs committed:
[refactor] Review 엔터티 @Getter 추가

변경 사항 :
 - Review 엔터티 @Getter 추가(피드백 반영)

ghrltjdtprbs committed:
[refactor] Tag엔터티 @Getter 추가

변경 사항 :
 - Tag 엔터티에 @Getter 추가하였습니다.(피드백 반영)

ghrltjdtprbs committed:
Merge branch 'develop' of https://github.com/fcstudy-project/FastTime-backend into feature/review

ghrltjdtprbs committed:
[feat] 중간 테이블 ReviewTag 생성

문제 사항 :
 - ManyToMany 사용 대신 중간 테이블 활용을 피드백 받음.
변경 사항 :
 - 중간 테이블 ReviewTag 생성
 - ManyToMany 삭제 및 수정

ghrltjdtprbs committed:
ìconflict 해결

ghrltjdtprbs commented:
> `@ManyToMany` 보다는 중간테이블을 하나 만드는게 더 좋아보이네요!

중간테이블 ReviewTag 생성해 커밋 했습니다! 확인해주세요

ghrltjdtprbs committed:
[feat] tagtype 추가

변경 사항 :
 - 긍정, 부정의 tagtype enum 추가

ghrltjdtprbs committed:
Merge branch 'develop' of https://github.com/fcstudy-project/FastTime-backend into feature/review

ghrltjdtprbs committed:
[refactor] tagtype 제거

문제 사항 :
 - 현재 상황에서 tagtype이 불필요하다는 피드백 수용.
 - rating 검증은 dto 에서 수행할 것을 제안
변경 사항 :
 - tagtype enum 제거
 - rating 검증 삭제

ghrltjdtprbs committed:
[refactor] Review 엔터티 검증 제거

문제 사항 :
 - rating 검증은 dto 에서 수행할 것을 제안
변경 사항 :
 - rating 검증 삭제

ghrltjdtprbs committed:
[feat] Review 도메인 Flyway 스크립트 추가

변경 사항 :
 - Review 도메인의 테이블(review, tag, reviewtag)생성을 위해  Flyway 스크립트를 추가하였습니다.
